### ISSUE TITLE
salt.modules.consul always serialize to JSON before storing into Consul's KV database

### ISSUE THREAD

**Issue by multani** [saltstack/salt#35215]
### Description of Issue/Question

The current `salt.modules.consul` module always serialize to JSON values which are stored into Consul's KV.
This is OK if we pass some structures to `consul.put` because it should only deals with strings, but if the value is already a string, then there's an unnecessary conversion to JSON which is done, which is specially a problem since the Consul's clients now have to always deserialize from JSON if they want to process the values set by the Salt module.
### Setup

```
# From Salt repository
$ ./scripts/salt-call consul.put consul_url=http://$CONSUL_HOST:$CONSUL_PORT key=/test/1 value=foo
$ ./bin/python -c "import consulate; print(repr(consulate.Consul('$CONSUL_HOST').kv.get('/test/1')))"
```

Produces the following:

```
$ ./scripts/salt-call consul.put consul_url=http://$CONSUL_HOST:$CONSUL_PORT key=/test/1 value=foo
[WARNING ] Failed to open log file, do you have permission to write to /var/log/salt/minion?
[WARNING ] Although 'dmidecode' was found in path, the current user cannot execute it. Grains output might not be accurate.
Cannot write to process directory. Do you have permissions to write to /var/cache/salt/minion/proc/20160804223620918076 ?
[ERROR   ] Key {0} does not exist. Skipping release.
local:
    ----------
    data:
        Added key /test/1 with value foo.
    res:
        True

$ ./bin/python -c "import consulate; print(repr(consulate.Consul('$CONSUL_HOST').kv.get('/test/1')))"
'"foo"'
```

So, it returns a string (which is expected), which contains a string encoded with JSON, although I would have expected to have only `"foo"` displayed.
### Versions Report

```
$ salt --versions-report
Salt Version:
           Salt: 2015.8.11

Dependency Versions:
         Jinja2: 2.7.3
       M2Crypto: Not Installed
           Mako: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.4.0
         Python: 2.7.9 (default, Mar  1 2015, 12:57:24)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: 0.8.6
       cherrypy: Not Installed
       dateutil: 2.2
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: Not Installed
      pycparser: 2.10
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: 0.8.2
        timelib: Not Installed

System Versions:
           dist: debian 8.5 
        machine: x86_64
        release: 3.16.0-4-amd64
         system: debian 8.5 
```

**Comment by Ch3LL** [comment]
ping @garethgreenaway can I get your input here? Looks like you created this module. It does indeed the data is being loaded as json. Think there is a better way to deal with this issue? Thanks

**Comment by multani** [comment]
I had a look recently with @cyrilgdn and it looks like that:

* no matter what gets passed to Consul, [it gets serialized as JSON](https://github.com/saltstack/salt/blob/develop/salt/modules/consul.py#L72)
* it makes sense for most of the methods which are passing structured data to modify Consul, using POST/PUT or DELETE methods
* it doesn't *always* make sense for the K/V interface, if the value is already a string.

We'll probably put up a patch in the coming days with a fix for this.

**Comment by djhaskin987** [comment]
Happening for me too :(
![image](https://user-images.githubusercontent.com/1186534/26851295-aa72e58e-4ac7-11e7-9fd6-f8e25a9d4532.png)

**Comment by Oloremo** [comment]
Run on this issue too. At make's close to zero sense to serialized data for KV as JSON. 
it's still not fixed as at 2018.3.3

**Comment by Oloremo** [comment]
@Ch3LL could this issue be reopened?

**Comment by Oloremo** [comment]
not stale

**Comment by cesan3** [comment]
This issue can be closed. It was fixed with commit 50a1743201. https://github.com/saltstack/salt/pull/61537.  Included in Phosphorus v3005.0
